### PR TITLE
EASY-2100: add the bagDirName to the deposit.properties

### PR DESCRIPTION
**Pull request by stephanHogenboom**
Fixes EASY-Easy 2100

#### When applied it will
*  add the bagDirName to the deposit.properties

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 1]
```diff
@@ -112,19 +109,12 @@ object DepositHandler extends BagValidationExtension {
...
+        FilesPermission.changePermissionsRecursively(depositDir, settings.depositPermissions, id).map(_ => receipt)
+      case f @ Failure(_) =>
+        FilesPermission.changePermissionsRecursively(depositDir, settings.depositPermissions, id).flatMap(_ => f)
     }
   }
 
-  private def setFilePermissions(depositDir: JFile)(implicit settings: Settings, id: DepositId): Try[Unit] = {
-    FilesPermission.changePermissionsRecursively(depositDir, settings.depositPermissions, id)
-      .doIfFailure {
-        case e: Exception => log.error(s"[$id] error while setting filePermissions for deposit: ${ e.getMessage }")
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:124]: You just lost this error log? Is that on purpose?

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 2]
```diff
@@ -112,19 +109,12 @@ object DepositHandler extends BagValidationExtension {
     // Otherwise operators don't have the proper permissions to clean or fix the invalid zip files or bags that might stay behind
     extractAndValidatePayloadAndGetDepositReceipt(deposit, contentLength, payload, depositDir) match {
       case Success(receipt) =>
-        setFilePermissions(depositDir).map(_ => receipt)
-      case Failure(exception) =>
-        setFilePermissions(depositDir).flatMap(_ => Failure(exception))
+        FilesPermission.changePermissionsRecursively(depositDir, settings.depositPermissions, id).map(_ => receipt)
+      case f @ Failure(_) =>
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala]:
```suggestion
      case f: Failure =>
```

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 3]
```diff
@@ -112,19 +109,12 @@ object DepositHandler extends BagValidationExtension {
     // Otherwise operators don't have the proper permissions to clean or fix the invalid zip files or bags that might stay behind
     extractAndValidatePayloadAndGetDepositReceipt(deposit, contentLength, payload, depositDir) match {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:110]: Now that you're at it, perhaps you can also get rid of the pattern match as well? and perhaps even extract the `FilesPermission.changePermissionsRecursively` into one call.

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 4]
```diff
@@ -180,6 +170,7 @@ object DepositHandler extends BagValidationExtension {
       _ <- checkBagVirtualValidity(bagDir)
       props <- DepositProperties(id)
       _ <- props.setState(SUBMITTED, "Deposit is valid and ready for post-submission processing")
+      _ <- props.setBagName(File(bagDir.getPath))
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala]: Just keep it easy: converting to better-files isn't really necessary here. `java.io.File` has something to get the name of a file as well.

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 5]
```diff
@@ -341,21 +332,19 @@ object DepositHandler extends BagValidationExtension {
...
-    try {
-      log.debug(s"[$id] Copying payload to: $zipFile")
-      Success(copyInputStreamToFile(deposit.getInputStream, zipFile))
-    } catch {
-      case t: Throwable => Failure(new SwordError(UriRegistry.ERROR_BAD_REQUEST, t))
-    }
+  def copyPayloadToFile(deposit: Deposit, zipFile: JFile)(implicit id: DepositId): Try[Unit] = Try {
+    log.debug(s"[$id] Copying payload to: $zipFile")
+    copyInputStreamToFile(deposit.getInputStream, zipFile)
+  } recover {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala]: Now you're losing an error...

```suggestion
  } recoverWith {
```

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 6]
```diff
@@ -341,21 +332,19 @@ object DepositHandler extends BagValidationExtension {
...
 
   def checkDepositIsInDraft(id: DepositId)(implicit settings: Settings): Try[Unit] = {
-    for {
-      props <- DepositProperties(id)
-      state <- props.getState
-      _ <- if (state == DRAFT) Success(())
-           else Failure(new SwordError(UriRegistry.ERROR_METHOD_NOT_ALLOWED, s"Deposit $id is not in DRAFT state."))
-    } yield ()
+    DepositProperties(id)
+      .flatMap(_.getState).flatMap {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala]: please format this better. second `flatMap` on new line and then auto-format.

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala (hunk 4 above)
Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala]: ok

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala (hunk 1 above)
Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:124]: i moved it to the filePermissions object

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala (hunk 3 above)
Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:110]: Deze snap ik niet helemaal

Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:110]: Let's leave it for now. I tried a bit, but it's not really worth it. I can't get rid of calling `changePermissionsRecursively` twice.

Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:110]: ok

**Pull request merged**